modules/cloudwatch_metrics/variables.tf

```diff
@@ -0,0 +1,79 @@
+variable "lambda" {
+  description = "Observe Lambda module"
+  type = object({
+    lambda_function = object({
+      arn  = string
+      role = string
+    })
+  })
+}
+
+variable "iam_name_prefix" {
+  description = "Prefix used for all created IAM roles and policies"
+  type        = string
+  nullable    = false
+  default     = ""
+}
+
+variable "statement_id_prefix" {
+  description = "Prefix used for Lambda permission statement ID"
+  type        = string
+  nullable    = false
+  default     = ""
+}
+
+variable "eventbridge_name_prefix" {
+  description = "Prefix used for eventbridge rule"
+  type        = string
+  nullable    = false
+  default     = "observe-lambda-metrics-"
+}
+
+variable "interval" {
+  description = "Interval in seconds between collection runs. Use a multiple of period to avoid gaps."
+  type        = number
+  nullable    = false
+  default     = 300
+  validation {
+    condition     = var.interval >= 60 && var.interval < 3600
+    error_message = "interval must be greater than or equal to 60 and less than 3600."
+  }
+}
+
+variable "period" {
+  description = "Period in seconds between metric data points. Must be a multiple of 60."
+  type        = number
+  nullable    = false
+  default     = 60
+  validation {
+    condition     = var.period >= 60 && var.period % 60 == 0
+    error_message = "period must be at least 60 seconds, and a multiple of 60."
+  }
+}
+
+variable "delay" {
+  description = "Collection delay in seconds. This delay accounts for the lag in metrics availability via Cloudwatch API."
+  type        = number
+  nullable    = false
+  default     = 300
+}
+
+variable "filters" {
+  description = "List of filters."
+  type = list(object({
+    namespace    = string
```

Comment:
is this required?

Reply:
Yes. The rationale here is that polling is a cost sensitive measure, and so the API should nudge people towards being specific. If you want a wildcard, you might as well configure a cloudwatch metrics stream.